### PR TITLE
Add support for other arch (x86 & ARM)

### PR DESCRIPTION
**Pull request by Edznux**
Hello !

I should have opened an issue first, but I wasn't expecting this to take that much time to add support for ARM and x86 arch.

The x86 support seems very similar to the x86_64 one (the output list is very similar).
The ARM one however seems to miss some syscall. (30 compared to 56 with x86_64 on go2seccomp itself) and I am not confident enough to say if this is the correct behaviour.

I have made some checks to ensure we are working on a Go binary. (with the `isGoBinary` function)

Some functions were moved to the "helpers.go" file. I can change the name or something else if you want :) (maybe put all the core code in a "lib" folder ? more like a library?)

As a beginner, I'm not sure if I should rebase this before the PR... So, just tell me and I will do it.

Thanks !

### CODE REVIEW

**Comment by xfernando** [comment]
Thanks Édouard! I'll try to review these later today! 😄

File: helpers.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+	switch file.Machine.String() {
+	case "EM_X86_64":
+		arch = specs.ArchX86_64
+	case "EM_386":
+		arch = specs.ArchX86
+	case "EM_ARM":
+		arch = specs.ArchARM
+	default:
+		log.Fatal("Unsuported arch : " + file.Machine.String())
```
Comment (xfernando) [inline helpers.go]: Missing newline, also maybe standardize to `Fatalf` and have error messages in this format "Something failed: %v\n", where "Something Failed" is whatever happened.

File: main.go [hunk 2]
```diff
@@ -109,60 +139,165 @@ func isSyscallPkgCall(instruction string) bool {
...
+
+	for i < previousInstructionsBufferSize {
+		instruction := previouInstructions[curPos%previousInstructionsBufferSize]
+		fmt.Println(instruction)
+		isMOVW := strings.Index(instruction, "MOVW") != -1
+		isBaseSPAddress := strings.Index(instruction, ", R0") != -1
+		// fmt.Println("isMOVW : ", isMOVW, "isBaseSPAddress : ", isBaseSPAddress)
+		syscallIDBeginning := strings.Index(instruction, "$")
+		if isMOVW && isBaseSPAddress && (syscallIDBeginning != -1) {
+			// if (syscallIDBeginning == -1 ){
```
Comment (xfernando) [inline main.go]: remove commented code

Comment (xfernando) [reply main.go]: if you want to keep debug level prints you can hide them behind the verbose flag

File: main.go [hunk 3]
```diff
@@ -109,60 +139,165 @@ func isSyscallPkgCall(instruction string) bool {
...
 
 	scanner := bufio.NewScanner(disassambled)
+
 	// keep a few of the past instructions in a buffer so we can look back and find the syscall ID
 	previousInstructions := make([]string, previousInstructionsBufferSize)
 	lineCount := 0
-	syscalls := getDefaultSyscalls()
+	syscalls := getDefaultSyscalls(arch)
+
+	// j := getCallOpByArch(arch)
```
Comment (xfernando) [inline main.go]: why is this commented out? it's the only place `getCallOpByArch()` is called

**Review by xfernando — COMMENTED** [review]
Just some minor comments, but a great job in general!

**Comment by Edznux** [comment]
Fixed :)
I failed to add `getCallOpByArch` when merging your upstream. That's better now : this add 5 syscalls on ARM (when testing on itself).
Thank you :smile:

**Pull request merged**